tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,524 @@
+        // "update-ssh-keys --add two valid2.pub"
+        test_add_two();
+
+        // "update-ssh-keys --delete two valid2.pub"
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid2", "two");
+    }
+
+    #[test]
+    fn test_disable() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --delete one valid1.pub"
+        // "update-ssh-keys --delete two valid2.pub"
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid1", "one");
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid2", "two");
+
+        // "update-ssh-keys --add one valid1.pub"
+        // "update-ssh-keys --add two valid2.pub"
+        test_add_two();
+
+        // "update-ssh-keys --disable two"
+        let sshkeyname1 = "valid1";
+        let sshkeyname2 = "valid2";
+
+        let mut aks = open_authorized_keys(ctx.ssh_dir.to_str().unwrap());
+        let keyfiles = [format!("{}/{}.pub", ctx.ssh_dir.to_str().unwrap(), sshkeyname2)];
+
+        let mut keys = vec![];
+        for keyfile in keyfiles {
+            let file = File::open(&keyfile).expect(format!("failed to open keyfile '{:?}'", keyfile).as_str());
+            keys.append(&mut AuthorizedKeys::read_keys(file).unwrap_or_default());
+        }
+
+        for key in aks.disable_keys("two") {
+            if let AuthorizedKeyEntry::Valid {
+                ref key,
+            } = key
+            {
+                assert!(key.to_fingerprint_string().contains(&*FINGERPRINTS[sshkeyname2]));
+
+                aks.write().expect("failed to update authorized keys directory");
+                aks.sync().expect("failed to update authorized keys");
+
+                let authkeyone: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/{}", ctx.ssh_dir.to_str().unwrap(), "two"));
+                assert!(authkeyone.exists());
+
+                let mut svec: Vec<&str> = Vec::new();
+                svec.push(sshkeyname1);
+                assert_has_keys(ctx.ssh_dir.to_str().unwrap(), svec, true);
+            }
+        }
+
+        // add two again
+        // "update-ssh-keys --add two valid2.pub"
+        //
+        // NOTE: Ideally adding two again here is supposed to immediately fail,
+        // but for some reason it simply hangs forever. For now do not run the
+        // following command to proceed to the next tests.
+
+        //        add_one_ssh_key(
+        //            ctx.ssh_dir.to_str().unwrap(), // ssh_dir
+        //            "valid2",                      // pubkeyname
+        //            "two",                         // testname
+        //            false,                         // is_force
+        //            true,                          // is_replace
+        //            ["valid1"].to_vec(),           // assert_keys
+        //        );
+    }
+
+    #[test]
+    fn test_enable() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --disable two"
+        test_disable();
```

Comment:
That's wrong because `test_disable` operates on its own private tmpdir, you should pass `ctx`.

Reply:
```suggestion
        test_disable_func(ctx);
```
Since it's also supposed to be an own test, you would have to add a wrapper for it
```
#[test]
fn test_disable() {
        let ctx = setup_tests();
        test_disable_func(ctx);
}
```

Reply:
Also, follow the same pattern for ` test_add_one_file` and `test_add_two`